### ISSUE TITLE
Serialiser is unable to serialise some container that it should serialise

### ISSUE THREAD

**Issue by phyBrackets** [fair-acc/opencmw-cpp#252]
While working on the DNS, I wanted to use the `std::set<std::string>` somewhere but it seems that the serialiser is unable to serialize and deserialise the `std::set` and not only with the `std::string`, other primitive types too. 
I'm not too sure that if the support for this exist or not but i guess it is, because it is mention in the [`CompileTimeSerialiser.md`](../tree/main/docs/CompileTimeSerialiser.md) doc. 
The reasons could be that the specialization for this type might be missing entirely, or that it's implemented incorrectly, or that the type itself is not supported by the IoSerializer (I might need to spend some time on the IoSerializer for the actual reason :) )
I also tried to use the some nested containers but probably that are not supported. :) 

Here are some error messages: 
```
/home/runner/work/opencmw-cpp/opencmw-cpp/src/serialiser/include/IoSerialiserYaS.hpp:281:55: error: ‘serialise’ is not a member of ‘opencmw::IoSerialiser<opencmw::YaS, std::set<std::__cxx11::basic_string<char> > >’
  281 |         IoSerialiser<YaS, StrippedDataType>::serialise(buffer, field, getAnnotatedMember(unwrapPointer(data)));
      |         ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~^~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~` 
 `/home/runner/work/opencmw-cpp/opencmw-cpp/src/serialiser/include/IoSerialiserCmwLight.hpp:275:60: error: ‘serialise’ is not a member of ‘opencmw::IoSerialiser<opencmw::CmwLight, std::set<std::__cxx11::basic_string<char> > >’
  275 |         IoSerialiser<CmwLight, StrippedDataType>::serialise(buffer, field, getAnnotatedMember(unwrapPointer(data)));
      |         ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~^~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
```

**Comment by wirew0rm** [comment]
It seems that they where meant to be implemented but never actually were. The line defining the type is [commented out](https://github.com/fair-acc/opencmw-cpp/blob/main/src/serialiser/include/IoSerialiserYaS.hpp#L61) and the specialisation is missing.

As for nested containers, I think the structure of the IoSerialiser has different codepaths for (reflectable) classes vs primitives and containers. From my understanding of the code `map<string, map<int, vector<double>>>` should work (but is definitely not well tested and might have bugs), but `map<int, my_refl_class>` will not.

I'll try to add the missing set case and some more tests for nested containers.

For the queue, the other missing type from the documentation I'm not sure if it should get it's own type id in the binary format or if it would just need specializations for the serialiser and share the binary format with array. (for json, set/queue/array/vector/[] would all map to `[a, b, c]` anyway, so the question is if the serialisation format should care about how the container behaves and not just what it stores. pro: kind of a type safety argument, con: might be valid for the server to store something in a queue and serialise that at some point, but the client might just want a vector of the entries). For now i'll just leave it out and mark it in the docs as unimplemented, i'd say.

Now that I think about it, the same argument could also be had for the map case, but for now i'd go with the way that is already laid out in the code unless anyone objects.

**Comment by phyBrackets** [comment]
Thanks @wirew0rm for a quick PR 😎.

**Comment by wirew0rm** [comment]
fixed by #253
